cmd/kubeadm/app/constants/constants_test.go

```diff
@@ -198,6 +198,61 @@ func TestEtcdSupportedVersion(t *testing.T) {
 	}
 }
 
+func TestEtcdNearestVersion(t *testing.T) {
+	var tests = []struct {
+		kubernetesVersion string
+		expectedVersion   *version.Version
+		expectedError     error
+	}{
+		{
+			kubernetesVersion: "1.10.0",
+			expectedVersion:   version.MustParseSemantic("3.2.24"),
+			expectedError:     nil,
+		},
+		{
+			kubernetesVersion: "1.15.1",
+			expectedVersion:   version.MustParseSemantic("3.3.10"),
+			expectedError:     nil,
+		},
+		{
+			kubernetesVersion: MinimumControlPlaneVersion.WithPatch(1).String(),
+			expectedVersion:   version.MustParseSemantic(SupportedEtcdVersion[uint8(MinimumControlPlaneVersion.Minor())]),
+			expectedError:     nil,
+		},
+		{
+			kubernetesVersion: CurrentKubernetesVersion.String(),
+			expectedVersion:   version.MustParseSemantic(SupportedEtcdVersion[uint8(CurrentKubernetesVersion.Minor())]),
+			expectedError:     nil,
+		},
+	}
+	for _, rt := range tests {
+		t.Run(rt.kubernetesVersion, func(t *testing.T) {
+			actualVersion, actualError := EtcdNearestVersion(rt.kubernetesVersion)
```

Comment:
i think we can avoid checking the .Error() string output of the expected error.

the checks here can be:
```
if (actualError != nil) != (rt.expectedError != nil) {
	t.Fatalf("expected error %v, got %v", rt.expectedError, actualError)
}
if actualError != nil {
	return
}
if actualVersion.String() != rt.expectedVersion.String() {
	t.Errorf("expected version %s, got %s", rt.expectedVersion.String(), actionVersion.String())
}
```